add modules to support pwm-fan

SmartElec opened the pull request:
userspace use command to enable pwn-fan
```
sudo sh -c 'echo XXX > /sys/devices/platform/pwm-fan/hwmon/hwmon1/pwm1' ## XXX : 范围 0-255 代表转速。
```

SmartElec committed:
Update rk3588-rock-5b.dts to support pwm fan

SmartElec committed:
Update rockchip_linux_defconfig

Update rockchip_linux_defconfig to support pwm-fan modules

CodeChenL commented:
设备树建议使用overlay😁

RadxaYuntian commented:
风扇接口是产品本身的一部分，从这个角度来看需要放到base dts里面。不过这个PR没有把风扇注册为thermal的cooling device。

SmartElec commented:
> 风扇接口是产品本身的一部分，从这个角度来看需要放到base dts里面。不过这个PR没有把风扇注册为thermal的cooling device。

注册测试了,不受控,温度达到了,风扇不转.cur_state一直为0,注册时和center_thermal绑定的.  就没加这部分进去了 😁

RadxaYuntian commented:
你要把cooling-levels还有cooling-cells写上去才行。可以参考[ZERO 2](https://github.com/radxa-repo/bsp/blob/main/linux/stable/0200-upstream/0004-arm64-dts-meson-add-support-for-Radxa-Zero2.patch#L76)，后面有添加fan0到CPU和RAM的cooling-device里面

SmartElec commented:
> 你要把cooling-levels还有cooling-cells写上去才行。可以参考[ZERO 2](https://github.com/radxa-repo/bsp/blob/main/linux/stable/0200-upstream/0004-arm64-dts-meson-add-support-for-Radxa-Zero2.patch#L76)，后面有添加fan0到CPU和RAM的cooling-device里面

加过了.官方默认策略是`power_allocator`,要配置策略为`step_wise`才生效. 内核里面只有一个宏`DEFAULT_THERMAL_GOVERNOR`来设置策略

SmartElec commented:
每个人的风扇的转速比不同,直接写死的话,也不能满足所有人.所以不太适合再这里引入coolingdevice

SmartElec committed:
Update rk3588-rock-5b.dts

register pwmfan to cooling device.
And user can use interface `/sys/class/thermal/cooling_device0/cur_state` to control
